uqcsbot/phonetics.py

```diff
@@ -0,0 +1,195 @@
+# length 4, 3, 2, and 1 for ease of decoding. This is not the most efficient way to
+# convert X-SAMPA to Unicode, but it is definitely the simplest + easiest to understand.
+
+XSAMPA_LOOKUP_4 = {
+    "G\\_<": "ʛ",
+    "J\\_<": "ʄ",
+    "|\\|\\": "ǁ",
+    "_B_L": " ᷅",
+    "_H_T": " ᷄",
+    "_R_F": " ᷈",
+}
+
+XSAMPA_LOOKUP_3 = {
+    "b_<": "ɓ",
+    "d_<": "ɗ",
+    "g_<": "ɠ",
+    "r\\`": "ɻ",
+    "<F>": "↘",
+    "<R>": "↗",
+}
+
+XSAMPA_LOOKUP_2 = {
+    "d`": "ɖ",
+    "h\\": "ɦ",
+    "j\\": "ʝ",
+    "l`": "ɭ",
+    "l\\": "ɺ",
+    "n`": "ɳ",
+    "p\\": "ɸ",
+    "r`": "ɽ",
+    "r\\": "ɹ",
+    "s`": "ʂ",
+    "s\\": "ɕ",
+    "t`": "ʈ",
+    "v\\": "ʋ",
+    "x\\": "ɧ",
+    "z`": "ʐ",
+    "z\\": "ʑ",
+    "B\\": "ʙ",
+    "G\\": "ɢ",
+    "H\\": "ʜ",
+    "I\\": "ᵻ",
+    "J\\": "ɟ",
+    "K\\": "ɮ",
+    "L\\": "ʟ",
+    "M\\": "ɰ",
+    "N\\": "ɴ",
+    "O\\": "ʘ",
+    "R\\": "ʀ",
+    "U\\": "ᵿ",
+    "X\\": "ħ",
+    "_j": "ʲ",
+    ":\\": "ˑ",
+    "@\\": "ɘ",
+    "@`": "ɚ",
+    "3\\": "ɞ",
+    "?\\": "ʕ",
+    "<\\": "ʢ",
+    ">\\": "ʡ",
+    "!\\": "ǃ",
+    "|\\": "ǀ",
+    "||": "‖",
+    "=\\": "ǂ",
+    "-\\": "‿",
+    '_"': " ̈",
+    "_+": " ̟",
+    "_-": " ̠",
+    "_/": " ̌",
+    "_0": " ̥",
+    "_=": " ̩",
+    "_>": "ʼ",
+    "_\\": " ̂",
+    "_^": " ̯",
+    "_}": " ̚",
+    "_~": " ̃",
+    "_A": " ̘",
+    "_a": " ̺",
+    "_B": " ̏",
+    "_c": " ̜",
+    "_d": " ̪",
+    "_e": " ̴",
+    "_f": " ̂",
+    "_G": "ˠ",
+    "_H": " ́",
+    "_h": "ʰ",
+    "_j": "ʲ",
+    "_k": " ̰",
+    "_L": " ̀",
+    "_l": "ˡ",
+    "_M": " ̄",
+    "_m": " ̻",
+    "_N": " ̼",
+    "_n": "ⁿ",
+    "_O": " ̹",
+    "_o": " ̞",
+    "_q": " ̙",
+    "_R": " ̌",
+    "_r": " ̝",
+    "_T": " ̋",
+    "_t": " ̤",
+    "_v": " ̬",
+    "_w": "ʷ",
+    "_X": " ̆",
+    "_x": " ̽",
+}
+
+XSAMPA_LOOKUP_1 = {
+    "A": "ɑ",
+    "B": "β",
+    "C": "ç",
+    "D": "ð",
+    "E": "ɛ",
+    "F": "ɱ",
+    "G": "ɣ",
+    "H": "ɥ",
+    "I": "ɪ",
+    "J": "ɲ",
+    "K": "ɬ",
+    "L": "ʎ",
+    "M": "ɯ",
+    "N": "ŋ",
+    "O": "ɔ",
+    "P": "ʋ",
+    "Q": "ɒ",
+    "R": "ʁ",
+    "S": "ʃ",
+    "T": "θ",
+    "U": "ʊ",
+    "V": "ʌ",
+    "W": "ʍ",
+    "X": "χ",
+    "Y": "ʏ",
+    "Z": "ʒ",
+    ".": ".",
+    '"': "ˈ",
+    "%": "ˌ",
+    "'": "ʲ",
+    ":": "ː",
+    "-": "",
+    "@": "ə",
+    "{": "æ",
+    "}": "ʉ",
+    "1": "ɨ",
+    "2": "ø",
+    "3": "ɜ",
+    "4": "ɾ",
+    "5": "ɫ",
+    "6": "ɐ",
+    "7": "ɤ",
+    "8": "ɵ",
+    "9": "œ",
+    "&": "ɶ",
+    "?": "ʔ",
+    "^": "ꜛ",
+    "!": "ꜜ",
+    "|": "|",
+    "~": " ̃",
+}
+
+
+class Phonetics(commands.Cog):
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    @app_commands.command(description="Converts X-SAMPA to IPA")
+    @app_commands.describe(input="X-SAMPA to convert")
+    async def xsampa(self, interaction: discord.Interaction, input: str):
+        remaining = input
```

Comment:
This isn't a change request, just a suggestion that you don't have to do. I should've mentioned it before but it would be good to have a simple python docstring briefly explaining the function. 

```python
async def xsampa(self, interaction: discord.Interaction, input: str):
    """
    Convert an X-SAMPA string to its equivalent IPA string. For example:

        /xsampa wVn drInk @nIw@n
    
        `wVn drInk @nIw@n`
        wʌn drɪnk ənɪwən
    """
    ...
``` 

As I said, this is just a suggestion to keep consistency among other functions in the codebase.

> **Note:** LGTM